### DISCUSSION TITLE
自动文档翻译怎么都触发不了，或者无法调用

### DISCUSSION THREAD

**Discussion by jiachangsheng** [YiiGuxing/TranslationPlugin#5849] [Ideas]
<img width="841" alt="rider64_blokD37GBy" src="https://github.com/YiiGuxing/TranslationPlugin/assets/50394909/c7244c6d-64e2-4faa-8e84-d97ea3959c0f">

**Comment by YiiGuxing** [comment]
请问您是怎么操作的？你指的自动翻译是指文档翻译吗？文档翻译是要在快速文档弹窗上进行的。

**Reply by jiachangsheng** [reply]
![image](https://github.com/user-attachments/assets/56ccb617-f0c6-43be-ba0e-9fb75f17f24f)
抱歉，之前没有表达清楚，两种方式都无法翻译，目前只能选词或者选择段落翻译

**Reply by jiachangsheng** [reply]
我在两台电脑上都试过了，都是无法翻译，而且快速文档弹窗上面也没有选项
![image](https://github.com/user-attachments/assets/967d0123-c852-445c-a37e-5de6d202ac99)

**Reply by YiiGuxing** [reply]
如果你使用的是 Rider 的话就不行， Rider 不支持文档翻译，只能通过选择翻译。

**Reply by jiachangsheng** [reply]
以后会有计划支持吗，我觉得这个功能对于阅读源码很有帮助，不过不能支持，那我就换个ide

**Reply by YiiGuxing** [reply]
是有计划的，只不过 Rider 是闭源的，不确定最终是不是能实现。

**Reply by miniscale** [reply]
@YiiGuxing 你好，HUAWEI DevEco Studio的文档也无法自动翻译，使用 @xuhuanzy 的分支可以自动翻译，它和这个是一样的问题吗。

**Reply by YiiGuxing** [reply]
> @YiiGuxing 你好，HUAWEI DevEco Studio的文档也无法自动翻译，使用 @xuhuanzy 的分支可以自动翻译，它和这个是一样的问题吗。

是的，目前这部分的支持正在开发中。

**Reply by miniscale** [reply]
好的，感谢回复。

**Comment by xuhuanzy** [comment]
分支支持了Rider: https://github.com/xuhuanzy/TranslationPlugin-New

**Reply by YiiGuxing** [reply]
还是只能用 `ja-netfilter` 吗？后面我研究一下能不能通过正规方式实现吧。你这个可以给急用的同学用用。

感谢！

**Reply by xuhuanzy** [reply]
代理的问题是插件的类加载器跟idea的类加载器不是同一个，代理挂上了找不到插件的类，给他加了个字段通过闭包的方式拿到了类加载器，结果打包后字段注入失败

**Reply by xuhuanzy** [reply]
> 还是只能用 `ja-netfilter` 吗？后面我研究一下能不能通过正规方式实现吧。你这个可以给急用的同学用用。

可以将翻译部分提取成Jar包, 然后再注入, 这样就不用 `ja-netfilter` 了, 但估计这工程量不小

**Reply by KaranocaVe** [reply]
感谢！可算能在rider里用文档翻译了
